### PR TITLE
Document composing filter conditions

### PR DESCRIPTION
**Pull request by raddevon**

### CODE REVIEW

**Review by scotttrinh — APPROVED** [review]
Love it!

**Review by elprans — COMMENTED** [review]
While you are at it, it might be worth talking about optionals and using `e.any` instead of a tree of `e.op(.. 'or' )`, because that question popped up recently.

**Comment by raddevon** [comment]
@scotttrinh Could you make one more pass on my additions per @elprans' suggestion? Added two more code blocks below what I had already added and brief explanations of each.

**Review by scotttrinh — COMMENTED** [review]
Might as well also mention `e.all` == `and` while we are at it.

**Comment by raddevon** [comment]
Good call, @scotttrinh. Done!

**Review by scotttrinh — APPROVED** [review]
Fabulous, thanks!!

File: docs/select.rst [hunk 1]
```diff
@@ -274,6 +274,132 @@ params object. This should correspond to a boolean expression.
...
+together with ``and``.
+
+.. code-block:: typescript
+
+  e.select(e.Movie, movie => ({
+    id: true,
+    title: true,
+    filter: e.all(
+      e.set(
+        e.op(movie.title, "ilike", "captain%")
```
Comment (themajashurka) [inline docs/select.rst:373]: Don't these lines need a comma at the end?

Comment (raddevon) [reply docs/select.rst:373]: Yes, I believe they do. Thank you for catching it and for letting us know!

Comment (raddevon) [reply docs/select.rst:373]: https://github.com/edgedb/edgedb-js/pull/744

**Pull request merged**